projects/example-app/src/app/auth/reducers/index.ts

```diff
@@ -32,7 +32,7 @@ export const selectAuthState = createFeatureSelector<State, AuthState>(
 
 export const selectAuthStatusState = createSelector(
   selectAuthState,
-  (state: AuthState) => state.status
+  (state: AuthState) => state[fromAuth.statusFeatureKey]
```

Comment:
Personally, I'm unsure if this adds value.
The difference between this selector and a `createFeatureSelector`, is that this selector is type-checked.

Reply:
Do you mean to remove the `selectAuthStatusState` selector completely and then change `selectUser` selector?

Reply:
No, not exactly. I only see a benefit of using feature keys for top-level slices (with `createFeatureSelector`). For selectors with the `createSelector` function, we have type safety in place and there are no magic strings because you can access the property (e.g. `state.status`). But this is my personal opinion 🙂

Reply:
Okey, now I see what you mean :smile: . I kind of agree 95% agree. One thing to add before dismissal would be,

* `state.status` would indeed be typechecked and give error for e.g. `state.notStatus`
* but for the other example,
```
// statusFeatureKey = 'status';

const randomValue = 'notStatus';

export interface AuthState {
  [fromAuth.statusFeatureKey]: fromAuth.State;
}

export const selectAuthStatusState = createSelector(
  selectAuthState,
  state => state[randomValue] <-- also gives type error 'notStatus' does not exist
);

// if randomValue instead has matching value
const randomValue = 'status';

state => state[randomValue] <-- this would be fine from type checking perspective
```

So the only thing here would be that if someone changes e.g. `statusFeatureKey` within the auth.reducer.ts there would be no need for direct changes if `state => state[fromAuth.statusFeatureKey]`.

However, I could see this as an invalid argue since the name of the selector presumably should be changed (selectAuth~~Status~~NewNameState).

I let you decide :relaxed:

Reply:
I preventively removed all the unnecessary things and squashed all the changes to one commit.

Reply:
Yes, you're right. 
(Also, the main IDEs can do a refactor to rename all occurrences of the property name)

We can wait for someone else his/her opinion before deciding.